frontend/catalyst/api_extensions/callbacks.py

```diff
@@ -256,8 +327,24 @@ def getOperandTypes(self):
         return list(map(ctypes.POINTER, operandTys))
 
 
+class JaxJitCallable(MemrefCallable):
+    """Callable that places the arguments in device before execution"""
+
+    def __init__(self, func, device, results_aval, *args, **kwargs):
+        assert device is not None, "Cannot have none device"
+        self.device = device
+        super().__init__(func, results_aval, *args, **kwargs)
+
+    def asarrays(self, void_ptrs):
+        """cast void_ptrs to jax arrays and move them to a device"""
+        expected_types = self.getOperandTypes()
+        jnparrays = MemrefCallable._asarrays(void_ptrs, expected_types)
+        movedarrays = [jax.device_put(array, self.device) for array in jnparrays]
```

Comment:
Are we sure this is needed? With `jax.jit` all values inside the function are already and exclusively on a given device.

Reply:
Oh my bad, just saw Josh's comment about `jax.jit(..., device=)` being deprecated

Reply:
Yes, it's a bit annoying (would have made the code much cleaner)